evm_loader/test_cancel.py

```diff
@@ -0,0 +1,170 @@
+from solana.transaction import AccountMeta, TransactionInstruction, Transaction
+from solana.rpc.types import TxOpts
+import unittest
+from base58 import b58decode
+from solana_utils import *
+from eth_tx_utils import make_keccak_instruction_data, make_instruction_data_from_tx
+from eth_utils import abi
+
+solana_url = os.environ.get("SOLANA_URL", "http://localhost:8899")
+http_client = Client(solana_url)
+CONTRACTS_DIR = os.environ.get("CONTRACTS_DIR", "evm_loader/")
+evm_loader_id = os.environ.get("EVM_LOADER")
+sysinstruct = "Sysvar1nstructions1111111111111111111111111"
+keccakprog = "KeccakSecp256k11111111111111111111111111111"
+sysvarclock = "SysvarC1ock11111111111111111111111111111111"
+
+
+class EventTest(unittest.TestCase):
```

Comment:
It is too wasteful to create a separate file to test each issue. Iterative execution tests are implemented in the `test_event.py` file. I suppose it is worth putting these tests there.